test/unit/sessions/test_attachment_models.py

```diff
@@ -0,0 +1,518 @@
+        assert worker_props.local_manifest_paths == provided_paths
+        # Verify it's the same reference (not a copy)
+        assert worker_props.local_manifest_paths is provided_paths
+
+    def test_local_manifest_paths_modification_after_initialization(self):
+        """Test that local_manifest_paths can be modified after initialization."""
+        manifest_props = ManifestProperties(
+            rootPath="/source/path", rootPathFormat=PathFormat.POSIX
+        )
+
+        worker_props = WorkerManifestProperties(
+            manifest_properties=manifest_props, local_root_path="/local/root"
+        )
+
+        # Start with empty list
+        assert worker_props.local_manifest_paths == []
+
+        # Add paths using various list methods
+        worker_props.local_manifest_paths.append("/manifest1.json")
+        assert worker_props.local_manifest_paths == ["/manifest1.json"]
+
+        worker_props.local_manifest_paths.extend(["/manifest2.json", "/manifest3.json"])
+        assert len(worker_props.local_manifest_paths) == 3
+        assert "/manifest2.json" in worker_props.local_manifest_paths
+        assert "/manifest3.json" in worker_props.local_manifest_paths
+
+        # Test insertion
+        worker_props.local_manifest_paths.insert(0, "/manifest0.json")
+        assert worker_props.local_manifest_paths[0] == "/manifest0.json"
+        assert len(worker_props.local_manifest_paths) == 4
+
+    def test_to_dict_serialization(self):
+        """Test converting WorkerManifestProperties to dictionary."""
+        manifest_props = ManifestProperties(
+            rootPath="/source/path",
+            rootPathFormat=PathFormat.POSIX,
+            fileSystemLocationName="shared_storage",
+            inputManifestPath="input.json",
+            inputManifestHash="hash123",
+            outputRelativeDirectories=["out1", "out2"],
+        )
+
+        worker_props = WorkerManifestProperties(
+            manifest_properties=manifest_props,
+            local_root_path="/local/root",
+            local_manifest_paths=["/local/manifest1.json", "/local/manifest2.json"],
+        )
+
+        result_dict = worker_props.to_dict()
+
+        # Verify structure
+        assert "manifestProperties" in result_dict
+        assert "localManifestPaths" in result_dict
+        assert "localRootPath" in result_dict
+        # Verify manifest field is not present
+        assert "manifest" not in result_dict
+
+        # Verify manifestProperties content
+        manifest_data = result_dict["manifestProperties"]
+        assert manifest_data["rootPath"] == "/source/path"
+        assert manifest_data["rootPathFormat"] == "posix"
+        assert manifest_data["fileSystemLocationName"] == "shared_storage"
+        assert manifest_data["inputManifestPath"] == "input.json"
+        assert manifest_data["inputManifestHash"] == "hash123"
+        assert manifest_data["outputRelativeDirectories"] == ["out1", "out2"]
```

Comment:
nit: we are now testing the behavior of `ManifestProperties.to_dict()` here.

Do we want to guard against changes to the `ManifestProperties` from `deadline-cloud`? Consider the case where a new field is added to it. The test would currently pass, but not have coverage for the new fields. Is that a problem?

Is it better to use a mock here if we defer to the behaviour of `ManifestProperties.to_dict(...)`?